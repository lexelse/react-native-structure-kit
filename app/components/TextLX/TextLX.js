// @flow

import * as React from "react";
import { Text } from "react-native";

type Props = {
    children?: React.Node
}

const TextLX = (props: Props) => {
    const { children } = props;

    return (
        <Text>{children}</Text>
    );
};

export default TextLX;
