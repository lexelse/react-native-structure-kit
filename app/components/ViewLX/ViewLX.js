// @flow

import * as React from "react";
import { View } from "react-native";

type Props = {
    children?: React.Node
}

const ViewLX = (props: Props) => {
    const { children } = props;
    return (
        <View>
            {children}
        </View>
    );
};

export default ViewLX;
