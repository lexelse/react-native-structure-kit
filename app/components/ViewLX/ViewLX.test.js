import React from "react";
import { shallow } from "enzyme";
import ViewLX from "./ViewLX";
import TextLX from "../TextLX/TextLX";

describe("ViewLX", () => {
    describe("rendering", () => {
        it("should match the snapshot", () => {
            const component = shallow(<ViewLX />).debug();
            expect(component).toMatchSnapshot();
        });

        describe("with props", () => {
            it("should render 'children' prop", () => {
                const component = shallow(<ViewLX><TextLX>HELLO WORLD</TextLX></ViewLX>).debug();
                expect(component).toMatchSnapshot();
            });
        });


        it.only("the data is peanut butter", () => {
            const mockCallBack = jest.fn();
            console.log(mockCallBack());

            mockCallBack.mockReturnValueOnce("first")
                .mockReturnValueOnce("second")
                .mockReturnValue(true);

            console.log(mockCallBack(), mockCallBack(), mockCallBack(), mockCallBack());
        });
    });
});

function forEach(items, callback) {
    for (let index = 0; index < items.length; index++) {
        callback(items[index]);
    }
}
