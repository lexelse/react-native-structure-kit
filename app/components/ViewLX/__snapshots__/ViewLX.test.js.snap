// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewLX rendering should match the snapshot 1`] = `"<View />"`;

exports[`ViewLX rendering with props should render 'children' prop 1`] = `
"<View>
  <TextLX>
    HELLO WORLD
  </TextLX>
</View>"
`;
